### ISSUE TITLE
Node stuck after resharding error

### ISSUE THREAD

**Issue by LeoHChen** [harmony-one/harmony#684]
**Describe the bug**

Reported by one of our node runners, the node program seems stuck after this logline at the end.

```
CRIT [04-05|22:27:38.520] CANNOT CALCULATE SHARDING STATE          port=9000 ip=18.228.138.169 epoch=4
```

Noted this is on banjo testnet.

**Comment by LeoHChen** [comment]
[harmony-18.228.138.169.log](https://github.com/harmony-one/harmony/files/3049888/harmony-18.228.138.169.log)

**Comment by rlan35** [comment]
This is related to the recent PR by Eugene adding error handling to Resharding. Resharding is not completed done and the PR added Crit log which may exit the program immediately. We should add exit program now as the resharding code is not fully integrated.

**Comment by LeoHChen** [comment]
Found another two occurrences in the recent launch of testnet.

**Comment by LeoHChen** [comment]
[validator-34.246.177.238-9000.log](https://github.com/harmony-one/harmony/files/3050957/validator-34.246.177.238-9000.log)

**Comment by harmony-ek** [comment]
@chaosma, the immediate thing we can do is to demote `Crit` to `Error`, which would let the program continue.

However, since `ss == nil`, it would later cause other panics at such places as `ss.shardState` and `ss.rnd` references and `ss.Reshard()` calls.  In order to avoid this, you would have to add a hack and synthesize an incompletely initialized/technically incorrect sharding state like:
```go
	if err != nil {
		utils.GetLogInstance().Error("CANNOT CALCULATE SHARDING STATE",
			"epoch", epoch)
		// TODO ek/rongjian/chao – DTRT instead of using an incomplete/incorrect sharding state.
		ss := &ShardingState{epoch: epoch} // all the other fields are zeroed out as would have been by the previous version of GetShardingStateFromBlockChain
	}

**Comment by chaosma** [comment]
I changed the log.Crit to log.Error and returned nil when calculate resharding state. We will wait for the whole resharding finished to solve it completely.

**Comment by harmony-ek** [comment]
Fix #700 merged.  Tracking the TODO item separately in #701.  Closing this.
